File: public/cubicle-claudedesign/ui_kits/ios/CubicleUI/Tokens.swift
// Cubicle Design System — iOS/SwiftUI Tokens
// Translated from tokens.css — follows HIG conventions

import SwiftUI

// ============================================================
// RAW PALETTE
// ============================================================
struct CP {
    static let n50   = Color(hex: 0xFAFAFA)
    static let n100  = Color(hex: 0xF5F5F5)
    static let n200  = Color(hex: 0xE5E5E5)
    static let n300  = Color(hex: 0xD4D4D4)
    static let n400  = Color(hex: 0xA3A3A3)
    static let n500  = Color(hex: 0x737373)
    static let n600  = Color(hex: 0x525252)
    static let n700  = Color(hex: 0x404040)
    static let n800  = Color(hex: 0x262626)
    static let n900  = Color(hex: 0x171717)
    static let n950  = Color(hex: 0x0A0A0A)
    static let red400   = Color(hex: 0xFF8787)
    static let red600   = Color(hex: 0xFA5252)
    static let green600 = Color(hex: 0x16A34A)
    static let amber600 = Color(hex: 0xD97706)
    static let white    = Color.white
    static let black    = Color.black
}

// ============================================================
// HEX INITIALIZER
// ============================================================
extension Color {
    init(hex: UInt, alpha: Double = 1) {
        self.init(
            .sRGB,
            red:   Double((hex >> 16) & 0xFF) / 255,
            green: Double((hex >>  8) & 0xFF) / 255,
            blue:  Double( hex        & 0xFF) / 255,
            opacity: alpha
        )
    }
}

// ============================================================
// SPACING
// ============================================================
struct CS {
    static let s3xs: CGFloat = 2
    static let s2xs: CGFloat = 4
    static let xs:   CGFloat = 8
    static let sm:   CGFloat = 12
    static let md:   CGFloat = 16
    static let lg:   CGFloat = 20
    static let xl:   CGFloat = 24
    static let xl2:  CGFloat = 32
    static let xl3:  CGFloat = 40
    static let xl4:  CGFloat = 48
    static let xl5:  CGFloat = 64
    static let xl6:  CGFloat = 80
}

// ============================================================
// RADII
// ============================================================
struct CR {
    static let none: CGFloat = 0
    static let xs:   CGFloat = 2
    static let sm:   CGFloat = 4
    static let md:   CGFloat = 6
    static let lg:   CGFloat = 8
    static let def:  CGFloat = 10
    static let xl:   CGFloat = 12
    static let xl2:  CGFloat = 16
    static let xl3:  CGFloat = 24
    static let full: CGFloat = 999
}

// ============================================================
// SEMANTIC THEME — Light + Dark (matches HIG adaptive colors)
// ============================================================
struct CT {
    let dark: Bool

    var bg:          Color { dark ? CP.black    : CP.white }
    var fg:          Color { dark ? CP.n50      : CP.n950 }
    var primary:     Color { dark ? CP.n100     : CP.n900 }
    var primaryFg:   Color { dark ? CP.n950     : CP.n50 }
    var secondary:   Color { dark ? CP.n800     : CP.n100 }
    var secondaryFg: Color { dark ? CP.n100     : CP.n900 }
    var muted:       Color { dark ? CP.n900     : CP.n100 }
    var mutedFg:     Color { dark ? CP.n400     : CP.n500 }
    var accent:      Color { dark ? CP.n900     : CP.n100 }
    var accentFg:    Color { dark ? CP.n100     : CP.n900 }
    var destructive: Color { dark ? CP.red400   : CP.red600 }
    var destructiveFg: Color { CP.white }
    var border:      Color { dark ? CP.n800     : CP.n200 }
    var input:       Color { dark ? CP.n800     : CP.n200 }
    var ring:        Color { dark ? CP.n400     : CP.n950 }
    var card:        Color { dark ? CP.n950     : CP.white }
    var cardFg:      Color { dark ? CP.n50      : CP.n950 }
    var green:       Color { CP.green600 }
    var amber:       Color { CP.amber600 }
    var overlay:     Color { Color.black.opacity(0.6) }
}

// ============================================================
// ENVIRONMENT KEY — pass CT through view tree
// ============================================================
private struct CubicleThemeKey: EnvironmentKey {
    static let defaultValue = CT(dark: false)
}

extension EnvironmentValues {
    var ct: CT {
        get { self[CubicleThemeKey.self] }
        set { self[CubicleThemeKey.self] = newValue }
    }
}

// ============================================================
// TYPOGRAPHY HELPERS — SF Pro with Cubicle proportions
// Uses system fonts for iOS native feel
// ============================================================
struct CTxt {
    static func sans(
        size: CGFloat = 15,
        weight: Font.Weight = .regular,
        design: Font.Design = .default
    ) -> Font {
        .system(size: size, weight: weight, design: design)
    }

    static func mono(size: CGFloat = 13, weight: Font.Weight = .regular) -> Font {
        .system(size: size, weight: weight, design: .monospaced)
    }

    // Preset styles matching mobile web kit
    static let appBarTitle  = Font.system(size: 17, weight: .semibold)
    static let largeTitle   = Font.system(size: 32, weight: .semibold)
    static let headline     = Font.system(size: 26, weight: .semibold)
    static let body         = Font.system(size: 15, weight: .regular)
    static let bodyMedium   = Font.system(size: 15, weight: .medium)
    static let callout      = Font.system(size: 14, weight: .regular)
    static let footnote     = Font.system(size: 13, weight: .regular)
    static let caption      = Font.system(size: 12, weight: .medium)
    static let captionMono  = Font.system(size: 11, weight: .medium, design: .monospaced)
    static let tabLabel     = Font.system(size: 10, weight: .medium)
    static let kpiValue     = Font.system(size: 26, weight: .bold)
    static let kpiLabel     = Font.system(size: 12, weight: .medium)
    static let bigNumber    = Font.system(size: 36, weight: .bold)
    static let chartBig     = Font.system(size: 24, weight: .bold)
}

// ============================================================
// SHADOW STYLES
// ============================================================
struct CShadow {
    static let sm = Shadow(color: .black.opacity(0.1),  radius: 2, x: 0, y: 1)
    static let md = Shadow(color: .black.opacity(0.1),  radius: 6, x: 0, y: 4)
    static let lg = Shadow(color: .black.opacity(0.12), radius: 15, x: 0, y: 10)
}

File: public/cubicle-claudedesign/ui_kits/flutter/lib/tokens.dart
// Cubicle Design System — Flutter Tokens
// Pixel-perfect translation from tokens.css

import 'package:flutter/material.dart';
import 'package:google_fonts/google_fonts.dart';

// ============================================================
// RAW COLOR PALETTE
// ============================================================
class CP {
  CP._();
  static const n50  = Color(0xFFFAFAFA);
  static const n100 = Color(0xFFF5F5F5);
  static const n200 = Color(0xFFE5E5E5);
  static const n300 = Color(0xFFD4D4D4);
  static const n400 = Color(0xFFA3A3A3);
  static const n500 = Color(0xFF737373);
  static const n600 = Color(0xFF525252);
  static const n700 = Color(0xFF404040);
  static const n800 = Color(0xFF262626);
  static const n900 = Color(0xFF171717);
  static const n950 = Color(0xFF0A0A0A);
  static const red400   = Color(0xFFFF8787);
  static const red600   = Color(0xFFFA5252);
  static const green600 = Color(0xFF16A34A);
  static const amber600 = Color(0xFFD97706);
  static const white    = Color(0xFFFFFFFF);
  static const black    = Color(0xFF000000);
}

// ============================================================
// SPACING — matches tokens.css exactly (in px = logical pixels)
// ============================================================
class CS {
  CS._();
  static const double s3xs = 2;
  static const double s2xs = 4;
  static const double xs   = 8;
  static const double sm   = 12;
  static const double md   = 16;
  static const double lg   = 20;
  static const double xl   = 24;
  static const double xl2  = 32;
  static const double xl3  = 40;
  static const double xl4  = 48;
  static const double xl5  = 64;
  static const double xl6  = 80;
}

// ============================================================
// BORDER RADII
// ============================================================
class CR {
  CR._();
  static const double none = 0;
  static const double xs   = 2;
  static const double sm   = 4;
  static const double md   = 6;
  static const double lg   = 8;
  static const double def  = 10;
  static const double xl   = 12;
  static const double xl2  = 16;
  static const double xl3  = 24;
  static const double full = 999;
}

// ============================================================
// SEMANTIC THEME TOKENS
// ============================================================
class CT {
  final bool dark;
  const CT(this.dark);

  Color get bg          => dark ? CP.black    : CP.white;
  Color get fg          => dark ? CP.n50      : CP.n950;
  Color get primary     => dark ? CP.n100     : CP.n900;
  Color get primaryFg   => dark ? CP.n950     : CP.n50;
  Color get secondary   => dark ? CP.n800     : CP.n100;
  Color get secondaryFg => dark ? CP.n100     : CP.n900;
  Color get muted       => dark ? CP.n900     : CP.n100;
  Color get mutedFg     => dark ? CP.n400     : CP.n500;
  Color get accent      => dark ? CP.n900     : CP.n100;
  Color get accentFg    => dark ? CP.n100     : CP.n900;
  Color get destructive => dark ? CP.red400   : CP.red600;
  Color get destructiveFg => CP.white;
  Color get border      => dark ? CP.n800     : CP.n200;
  Color get input       => dark ? CP.n800     : CP.n200;
  Color get ring        => dark ? CP.n400     : CP.n950;
  Color get card        => dark ? CP.n950     : CP.white;
  Color get cardFg      => dark ? CP.n50      : CP.n950;
  Color get green       => CP.green600;
  Color get amber       => CP.amber600;
  Color get overlay     => const Color(0x99000000);

  BoxShadow get shadowSm => BoxShadow(
    color: Colors.black.withOpacity(dark ? 0.3 : 0.1),
    blurRadius: 3,
    offset: const Offset(0, 1),
  );
  BoxShadow get shadowMd => BoxShadow(
    color: Colors.black.withOpacity(dark ? 0.3 : 0.1),
    blurRadius: 6,
    offset: const Offset(0, 4),
  );
  BoxShadow get shadowLg => BoxShadow(
    color: Colors.black.withOpacity(dark ? 0.35 : 0.1),
    blurRadius: 15,
    offset: const Offset(0, 10),
  );
}

// ============================================================
// TEXT STYLES — Geist via Google Fonts
// ============================================================
class CTxt {
  CTxt._();

  static TextStyle sans({
    double size = 15,
    FontWeight weight = FontWeight.w400,
    double? lineHeight,
    double? letterSpacing,
    Color? color,
  }) =>
      GoogleFonts.geist(
        fontSize: size,
        fontWeight: weight,
        height: lineHeight != null ? lineHeight / size : null,
        letterSpacing: letterSpacing,
        color: color,
      );

  static TextStyle mono({
    double size = 13,
    FontWeight weight = FontWeight.w400,
    double? lineHeight,
    double? letterSpacing,
    Color? color,
  }) =>
      GoogleFonts.geistMono(
        fontSize: size,
        fontWeight: weight,
        height: lineHeight != null ? lineHeight / size : null,
        letterSpacing: letterSpacing,
        color: color,
      );
}

// ============================================================
// INHERITED THEME WIDGET — pass CT down the tree
// ============================================================
class CubicleTheme extends InheritedWidget {
  final CT t;
  const CubicleTheme({super.key, required this.t, required super.child});

  static CT of(BuildContext context) {
    return context.dependOnInheritedWidgetOfExactType<CubicleTheme>()!.t;
  }

  @override
  bool updateShouldNotify(CubicleTheme old) => old.t.dark != t.dark;
}
